File: webui_app/api/v1/schedule.py
"""Scheduled-drafts read endpoint for ``/api/v1`` — Plan 2026-06-18-002 U7.

Migrates the ``/schedule`` page (a read-only table of drafts with a future
``scheduled_at``) off the legacy Jinja route + ``/api/scheduled`` ESM fetch.
REUSES the existing ``scheduled_api.list_scheduled`` query — no new facade; the
schedule view and the drafts queue stay a single source over ``drafts_store``.

Read-only + fail-soft: a query failure degrades to an empty list (the SPA shows
the empty state), mirroring the legacy route's ``ok``-flag fallback rather than
surfacing a transport error for a pure listing.
"""

from __future__ import annotations

from flask import jsonify

from ..scheduled_api import list_scheduled
from . import bp


@bp.get("/schedule")
def schedule_list():
    """Drafts scheduled for future publish (newest-config order)."""
    result = list_scheduled()
    return jsonify({"items": result.get("items", [])})

File: frontend/src/api/schedule.ts
// Typed wrapper for the /api/v1/schedule endpoint (Plan 2026-06-18-002 U7).
// Read-only view of drafts scheduled for future publish — single source over
// drafts_store, shared with the drafts queue (mutations live on the Drafts page).

import { getJson } from './client'

export interface ScheduledItem {
  id?: string
  title?: string
  target_url?: string
  platform?: string
  scheduled_at?: string | null
  created_at?: string
  status?: string
}

export interface ScheduledList {
  items: ScheduledItem[]
}

export const listScheduled = (): Promise<ScheduledList> => getJson('/schedule')
